### DISCUSSION TITLE
best approach for fast consumers

### DISCUSSION THREAD

**Discussion by masouddarvishi** [OptimalBits/bull#2382] [General]
hello every one

guys I haven been searching about this for days but there wasn't any answer I can work with (or should I say did not understand it), plz help me:

let's assume there is queue named "notification-queue" and there is a lot of jobs inside that queue. we want to process them as fast as possible. here is my questions:

1 - should I use consumer with increased concurrency number something like 10 or 20?
2 - should I use named jobs with concurrency 1? (one consumer with a process method for each named job)
3 - should I use multi consumers with concurrency 1? (if you suggest this please provide me link or some thing so I can work with I couldn't find anything)

and there is something that I can't understand it :
when we increase concurrency of consumer what exactly means? for example if concurrency of consumer has been set to 2, does it mean there is actual 2 instance of that consumer? or it means there is just one consumer but processing two jobs at same time? which one is faster? having multi consumer or one consumer with increased concurrency. 

thank you  in advance.

**Comment by manast** [comment]
Let me know if any of your questions are not being answered here: https://docs.bullmq.io/guide/workers/concurrency
